site/docs/plugins/i18n.md

```diff
@@ -1,105 +1,694 @@
 
-## Usage
+## Getting Started
+
+There are [multiple ways](#adding-translations) to add more languages to your bot.
+The easiest way is creating a folder with your fluent translation files in it.
+Usually, the name of that folder is going to be `locales/`.
+The translation files should have the extension `.ftl` (fluent).
 
-The basic thing you need are the translations.
-Normally they are stored in your bot structure as yaml files (but this plugin supports other ways to handle them.)
+Here is an example project structure:
 
-```plaintext
-yaml and json are ok
-Example directory structure:
-├── locales
-│   ├── en.yaml
-│   ├── en-US.yaml
-│   ├── it.json
-│   └── ru.yaml
-└── bot.js
+```asciiart:no-line-numbers
+.
+├── bot.ts
+└── locales/
+    ├── de.ftl
+    ├── en.ftl
+    ├── it.ftl
+    └── ru.ftl
 ```
 
-These files contain translations like this:
+If you're unfamiliar with the Fluent file (`.ftl`) syntax, you can have a look at their syntax guide here: <https://projectfluent.org/fluent/guide>
+
+These translation files contain translations like this:
 
-en.yaml
+`en.ftl`
 
-```yaml
-hey: Hello!
-help:
-  short: This bot can help you with stuff.
-  long: >
-    This bot can help you with stuff.
-    For example…
+```fluent
+start = Hi, how can I /help you?
+help = 
+    Send me some text and I can make it bold for you.
+    You can change my language using the /language command.
 ```
 
-de.yaml
+`de.ftl`
 
-```yaml
-hey: Hallo!
-help:
-  short: Dieser Bot kann dir bei Dingen helfen.
-  long: >
-    Dieser Bot kann dir bei Dingen helfen.
-    Zum Beispiel…
+```fluent
+start = Hallo, wie kann ich dir helfen? /help
+help = 
+    Schick eine Textnachricht, die ich für dich fett schreiben soll.
+    Du kannst mit dem Befehl /language die Spache ändern.
 ```
 
-As you can see the same keys are existing in both files.
-When we now want to send a message to a user with have to use the key.
+Using the translations in your bot:
 
 ```ts
 bot.command("start", async (ctx) => {
-  await ctx.reply(ctx.i18n.t("hey"));
+  await ctx.reply(ctx.t("start"));
+});
+
+bot.command("help", async (ctx) => {
+  await ctx.reply(ctx.t("help"));
 });
 ```
 
-Depending on the user language the Bot will now reply with a different text.
+Whenever a message is sent the locale of context is used. This could be your specific language negotiator or just `ctx.from.language_code`
+So as a result different users with different locales will be able to read the messages in their languages.
+
+## Usage
+
+The plugin chooses the language in use by different ways. One of them is the default `ctx.from.language_code` provided by the telegram client of the user.
+If your bot has a command for changing the language it is better to use the language chosen by the user.
+Based on these different language selection approaches the i18n plugin can be configured mainly in two different ways: [With sessions](#with-sessions) or [Without sessions](#without-sessions).
```

Comment:
I like how this was improved, but I still see some potential here to express this idea in a more structured way. We effectively mean is that the language is determined by different factors, and one of these factors is whether we have access to sessions. Hence, there are two ways to use the plugin. LMK if you don't see a way yourself how this can be improved, I can offer to rephrase this paragraph.